### ISSUE TITLE
[SF]Icon: A11y - Icon Cannot Be Activated With Enter and Space Key in JAWS/NVDA

### ISSUE THREAD

**Issue by VidhiNPatel** [SAP/ui5-webcomponents-react#7293]
### Describe the bug

The Icon Cannot Be Activated With the Enter and Space key in "Interactive" mode.
Keyboard Handling is not working in JAWS/ NVDA screen reader
[Space] / [Enter] or [Return] - Fires the click event if the mode property is set to Interactive.
https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/data-display-icon--docs 

### Reproduction steps

1. Go to https://stackblitz.com/edit/github-t5j5mn-u4rgcw6c?file=src%2FApp.tsx,src%2Findex.css,package.json 
2. Start the JAWS/NVDA screen reader
3. Use [Space] / [Enter] or [Return] to click on the Icon
4. Observed that the Icon Cannot Be Activated With Enter and Space Key

### UI5 Web Components for React Version

2.9.2

### UI5 Web Components Version

2.9.0

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.
